### PR TITLE
Exact Size

### PR DESCRIPTION
**Pull request by felipetnh**
@adityapatadia

### CODE REVIEW

**Comment by felipetnh** [comment]
@peter279k

File: lib/ImageResize.php [hunk 1]
```diff
@@ -105,7 +105,7 @@ public function __construct($filename)
         if (!defined('IMAGETYPE_WEBP')) {
             define('IMAGETYPE_WEBP', 18);
         }
-        if ($filename === null || empty($filename) || (substr($filename, 0, 5) !== 'data:' && !is_file($filename))) {
+        if ($filename === null || empty($filename) || (substr($filename, 0, 7) !== 'data://' && !is_file($filename))) {
```
[the remark this answers is not in the document]
Comment (felipetnh) [reply lib/ImageResize.php]: This substr($filename, 0, 7) was on the master branch. The version I downloaded was the previous one. So I wasn't sure what to do.

File: lib/ImageResize.php [hunk 2]
```diff
@@ -229,41 +234,70 @@ public function save($filename, $image_type = null, $quality = null, $permission
...
             if (!$this->quality_truecolor && !imageistruecolor($this->source_image)) {
-                $dest_image = imagecreate($this->getDestWidth(), $this->getDestHeight());
-
-                $background = imagecolorallocatealpha($dest_image, 255, 255, 255, 1);
-                imagecolortransparent($dest_image, $background);
-                imagefill($dest_image, 0, 0, $background);
+                if( !empty($exact_size) && is_array($exact_size) ){
+                    $dest_image = imagecreate($exact_size[0], $exact_size[1]);
+                } else{
+                    $dest_image = imagecreate($this->getDestWidth(), $this->getDestHeight());
```
[the remark this answers is not in the document]
Comment (peter279k) [reply lib/ImageResize.php]: The white space seems to be existed before the `$dest_image` variable :).

File: lib/ImageResize.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (peter279k) [reply lib/ImageResize.php]: Well. Checking the latest merged [PR ](https://github.com/gumlet/php-image-resize/pull/119/files)seems that it's different.

[the remark this answers is not in the document]
Comment (peter279k) [reply lib/ImageResize.php]: The master branch is `($filename, 0, 5)`. Please check [that](https://github.com/gumlet/php-image-resize/blob/master/lib/ImageResize.php#L108) :-).

**Comment by adityapatadia** [comment]
@felipetnh if you can fix the CI error, we can merge this.

**Comment by peter279k** [comment]
@adityapatadia, it seems that the CI work is successful, but the `coveralls` is failed.

It looks like the code coverage is decreased and I think we need to add more tests to recover the code coverage...

**Comment by felipetnh** [comment]
Guys, I have no idea what you're talking about.
CI errors? Coveralls?
Those things aren't really my thing.
Tell me what I need to do and I'll gladly update the PR.

Thanks!


Em qui, 4 de out de 2018 04:33, peter279k <notifications@github.com>
escreveu:

**Comment by peter279k** [comment]
Hi @felipetnh, do you see this captured image?
image

The `coverage/coveralls` is failed because the coverage decreased. You need to do is add more tests to recover the code coverage back :smile:.

**Comment by felipetnh** [comment]
And how do I do that?
I've never worked with test cases.

Em qui, 4 de out de 2018 às 07:48, peter279k <notifications@github.com>
escreveu:

**Comment by adityapatadia** [comment]
It would be great if we can also add details about this in Readme.

**Comment by peter279k** [comment]
Hi @adityapatadia, how about adding some details in released draft when releasing the latest version?

**Comment by ekinhbayar** [comment]
Any update on this?

**Comment by peter279k** [comment]
Hi @ekinhbayar, thank you for your concern.

Just wait for @adityapatadia review this PR.

File: lib/ImageResize.php [hunk 3]
```diff
@@ -229,41 +234,70 @@ public function save($filename, $image_type = null, $quality = null, $permission
...
+            imagefill($dest_image, 0, 0, $background);
             break;
         }
 
         imageinterlace($dest_image, $this->interlace);
-        
+
         imagegammacorrect($this->source_image, 2.2, 1.0);
+
+        if( !empty($exact_size) && is_array($exact_size) ) {
```
Comment (peter279k) [inline lib/ImageResize.php]: It mentions on issue #142.

**Comment by adityapatadia** [comment]
I would still appreciate update in Readme.

File: lib/ImageResize.php (hunk 3 above)
Comment (peter279k) [reply lib/ImageResize.php]: @adityapatadia, this codes will effect and cause on the issue #142.

The line `291` to `300`.

**Pull request merged**